getwax/wax
# 136 run all integration tests against bundler

**What has this PR changed:**
- Assumes `addRecoveryAccount` is called from a safe, this is the assumption on how the owner would normally call this function through a userOp. Maybe we also want to cater to the assumption that `addRecoveryAccount` could be called directly by the owner (see comment).
- Runs all integration tests through a bundler 
  - Bls plugin and recovery plugin didn't do this before 
- Shares `setupTests` logic in dedicated file
- Uses an `admin` account for actions such as deploying contracts
- Wraps admin account in an ethers `NonceManager` to prevent several `nonce too low` errors.
  - Also derives second account from mnemonic to avoid clashes with hardhat account 0 which is default account the eth-infinitism bundler uses
- Moves gas estimation logic to dedicated function - `getGasEstimates()`
- Moves logic to create user ops to dedicated file - `createUserOp.ts`

**How to test these changes:**
1. Check env vars are correct according to `.env.example`
2. run `./script.start.sh` in `account-integrations/safe`
3. run `yarn run bundler` in the eth infinitism bundler at commit `1b154c9`
4. Run `yarn hardhat test` in `account-integrations/safe`
